pkgs/stdenv/darwin/default.nix

```diff
@@ -314,403 +308,604 @@ rec {
-          gnumake gzip gnused bzip2 gawk ed xz patch bash
-          ncurses libffi zlib gmp pcre gnugrep
-          coreutils findutils diffutils patchutils pbzx;
+      persistent = self: super: {
+        inherit (prevStage)
+          bash binutils brotli bzip2 coreutils curl diffutils ed file findutils gawk gettext
+          gmp gnugrep gnumake gnused gnutar gzip libffi libidn2 libiconv libssh2
+          libunistring libxml2 ncurses nghttp2 openssl patch pre xz zlib zstd;
 
         darwin = super.darwin.overrideScope (_: _: {
-          inherit (darwin) dyld ICU Libsystem Csu libiconv rewrite-tbd;
+          inherit (prevStage.darwin) dyld Libsystem CF ICU locale libtapi;
         } // lib.optionalAttrs (super.stdenv.targetPlatform == localSystem) {
-          inherit (darwin) binutils binutils-unwrapped cctools;
+          inherit (prevStage.darwin) binutils binutils-unwrapped;
         });
       } // lib.optionalAttrs (super.stdenv.targetPlatform == localSystem) {
-        inherit llvm;
+        inherit (prevStage) llvm;
 
         # Need to get rid of these when cross-compiling.
-        "${finalLlvmPackages}" = super."${finalLlvmPackages}" // (
+        llvmPackages = super.llvmPackages // (
           let
-            tools = super."${finalLlvmPackages}".tools.extend (_: super: {
-              inherit (pkgs."${finalLlvmPackages}") llvm clang-unwrapped;
+            tools = super.llvmPackages.tools.extend (_: _: {
+              inherit (prevStage.llvmPackages) clang clang-unwrapped libclang libllvm llvm;
             });
-            libraries = super."${finalLlvmPackages}".libraries.extend (_: _: {
-              inherit (pkgs."${finalLlvmPackages}") compiler-rt libcxx libcxxabi;
+            libraries = super.llvmPackages.libraries.extend (_: _: {
+              inherit (prevStage.llvmPackages) compiler-rt libcxx libcxxabi;
             });
           in
           { inherit tools libraries; } // tools // libraries
         );
 
-        inherit binutils binutils-unwrapped;
+        inherit (prevStage) binutils binutils-unwrapped;
       };
+
+      cc = prevStage.llvmPackages.clang;
     in
-    import ../generic rec {
+    {
+    inherit config overlays;
+    stdenv = import ../generic {
       name = "stdenv-darwin";
 
-      inherit config;
-      inherit (pkgs.stdenv) fetchurlBoot;
-
       buildPlatform = localSystem;
       hostPlatform = localSystem;
       targetPlatform = localSystem;
 
-      preHook = commonPreHook + ''
-        export NIX_COREFOUNDATION_RPATH=${pkgs.darwin.CF}/Library/Frameworks
-        export PATH_LOCALE=${pkgs.darwin.locale}/share/locale
-      '';
-
-      __stdenvImpureHostDeps = commonImpureHostDeps;
-      __extraImpureHostDeps = commonImpureHostDeps;
+      inherit config;
 
-      initialPath = import ../generic/common-path.nix { inherit pkgs; };
-      shell = "${pkgs.bash}/bin/bash";
+      preHook = commonPreHook + ''
+          export PATH_LOCALE=${prevStage.darwin.locale}/share/locale
```

Comment:
Why not guard `PATH_LOCALE` by the `targetPlatform` check as well?

Reply:
It didn’t occur to me at the time, but I’ve fixed it in my local branch. I’ll be pushing an update to move `PATH_LOCALE` under the `targetPlatform` check and removing the comment later today (hopefully after I resolve the issue with an infinite recursion building xnu on x86_64-darwin).